tests/utils/component_test_utils.go

```diff
@@ -130,42 +166,45 @@ func setContainerComponentValues(containerComponent *schema.ContainerComponent)
 		containerComponent.Env = nil
 	}
 
-	if GetBinaryDecision() {
-		containerComponent.VolumeMounts = addVolume(GetRandomNumber(4))
-	} else {
-		containerComponent.VolumeMounts = nil
+	if len(containerComponent.VolumeMounts) == 0 {
+		if GetBinaryDecision() {
+			containerComponent.VolumeMounts = devfile.addVolume(GetRandomNumber(4))
+		}
 	}
 
 	if GetBinaryDecision() {
-		containerComponent.Endpoints = CreateEndpoints()
+		containerComponent.Endpoints = devfile.CreateEndpoints()
```

Comment:
Are we going  to test component with multiple endpoints in the future?
Just to note that, the port defined in endpoints has to be unique across components, but same container can have multiple endpoints sharing  the same port.  

So the restriction looks correct in the current test coverage, but are we looking to expend the coverage?

Reply:
Test already includes multiple endpoints. Updated the test so that it will randomly use the same port when adding multiple endpointd to a given container.